### PR TITLE
feat: add dual db warm up support for blocksyncer

### PR DESCRIPTION
**Pull request by krish-nr**
### Description

add dual db warm up support for block syncer

### Rationale

each time we release a new version, block syncer need lot of time to get the latest block height. So we init 2 db when blocksyncer starts and switch master-slave relation when new db get the latest height

### Example

N/A

### Changes

Notable changes: 
* init 2 block syncer service when start up
* DB switch when new DB is ready
* upgrade juno version for new configs

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -130,7 +130,9 @@ var DefaultGreenfieldChainConfig = &gnfd.GreenfieldChainConfig{
 var DefaultBlockSyncerConfig = &blocksyncer.Config{
 	Modules:        []string{"epoch", "bucket", "object", "payment"},
 	Dsn:            "localhost:3308",
-	RecreateTables: true,
+	DsnBackup:      "localhost:3308",
```
Comment (will-2012) [inline config/config.go:133]: BackupDsn?

File: config/config.go [hunk 2]
```diff
@@ -130,7 +130,9 @@ var DefaultGreenfieldChainConfig = &gnfd.GreenfieldChainConfig{
...
+	RecreateTables: false,
+	Backup:         false,
```
Comment (will-2012) [inline config/config.go:135]: What is the meaning? `EnableBackup`?

Comment (will-2012) [reply config/config.go:135]: Update config/config_template.toml synchronously

File: model/const.go [hunk 3]
```diff
@@ -121,6 +123,8 @@ const (
 	SpSealPrivKey = "SIGNER_SEAL_PRIV_KEY"
 	// DsnBlockSyncer defines env variable name for block syncer dsn
 	DsnBlockSyncer = "BLOCK_SYNCER_DSN"
+	// DsnBlockSyncerBackup defines env variable name for block syncer backup dsn
+	DsnBlockSyncerBackup = "BLOCK_SYNCER_DSN_SWITCHED"
```
Comment (will-2012) [inline model/const.go:127]: Why not keep SWITCHED and backup in the same naming style?

File: service/blocksyncer/block_syncer.go [hunk 4]
```diff
@@ -178,36 +282,45 @@ func (s *BlockSyncer) serve(ctx context.Context) {
...
+		select {
+		case <-ctx.Done():
+			log.Infof("Receive cancel signal, getLatestBlockHeight routine will stop")
+			return
+		default:
+			{
+				latestBlockHeight, err := s.parserCtx.Node.LatestHeight()
+				if err != nil {
+					log.Errorw("failed to get last block from RPCConfig client",
+						"err", err,
```
Comment (will-2012) [inline service/blocksyncer/block_syncer.go:304]: `"error"`

File: service/blocksyncer/block_syncer.go [hunk 5]
```diff
@@ -35,27 +42,94 @@ var (
...
+	MainService   *BlockSyncer
+	BackupService *BlockSyncer
+
+	FlagDB database.Database
+
+	NeedBackup bool
+
+	CancelMain func()
+	CtxMain    context.Context
 )
```
Comment (will-2012) [inline service/blocksyncer/block_syncer.go:54]: Maybe in the future it can be encapsulated inside the module and switched internally.

**Review by joeylichang — APPROVED** [review]

File: service/blocksyncer/block_syncer_service.go [hunk 6]
```diff
@@ -43,3 +54,61 @@ func mustGetLatestHeight(ctx *parser.Context) uint64 {
...
+	masterFlag.IsMaster = !masterFlag.IsMaster
+	if err = FlagDB.SetMasterDB(context.TODO(), masterFlag); err != nil {
+		return err
+	}
+	log.Infof("DB switched")
+	return nil
+}
+
+func determineMainService() error {
+	masterFlag, err := FlagDB.GetMasterDB(context.TODO())
```
Comment (reneecok) [inline service/blocksyncer/block_syncer_service.go:101]: when masterFlag is nil?

**Review by constwz — APPROVED** [review]

File: go.mod [hunk 7]
```diff
@@ -1,6 +1,6 @@
 module github.com/bnb-chain/greenfield-storage-provider
 
-go 1.20
+go 1.18
```
Comment (will-2012) [inline go.mod:3]: Pls, check the Golang version~

Comment (2129zxl) [reply go.mod]: resolved，thx

**Pull request merged**